compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -569,6 +569,11 @@ impl<'p, 'tcx> MatchVisitor<'p, 'tcx> {
     ) {
         assert!(self.let_source != LetSource::None);
 
+        // for let chains, don't emit IRREFUTABLE_LET_PATTERNS
+        if chain_refutabilities.len() > 1 {
+            return;
+        }
+
         if chain_refutabilities.iter().all(|r| matches!(*r, Some((_, Irrefutable)))) {
             // The entire chain is made up of irrefutable `let` statements
             report_irrefutable_let_patterns(
```

Comment:
It seems the check you're adding should cause a reworking of the logic that follows.